gm4_blossoming_pots/data/gm4_blossoming_pots/function/flower/as_player.mcfunction

```diff
@@ -0,0 +1,14 @@
+# Sets up raycast, then redirects to as_marker, then revokes advancement
+# @s = player who interacted with an empty flower pot
+# at @s
+# run from advancement gm4_blossoming_pots:interact_flower_pot
+
+# Raycast for the pot (set up range and search)
+execute store result score @s gm4_blossoming_pots.range run attribute @s minecraft:block_interaction_range get 200
+execute anchored eyes run function gm4_blossoming_pots:flower/raycast/loop
+
+execute as @e[type=minecraft:marker,sort=nearest,tag=gm4_blossoming_pots.temp.flower_pot] at @s \
+    align xyz positioned ~.5 ~.5 ~.5 run function gm4_blossoming_pots:flower/as_marker with entity @s data
+
+# Reset advancement
+advancement revoke @s only gm4_blossoming_pots:interact_flower_pot
```

Comment:
Probably best to do this first, just in case there is some runtime error causing the function to abort prematurely